src/modules/party/dto/get-parties-by-search-text-query.dto.ts

```diff
@@ -1,6 +1,6 @@
 import { ValidatedStringApiProperty } from '@ukef/decorators/validated-string-api-property.decorator';
 
-const stringWithoutWhitespaces = /^\s*\S+\s*$/;
+const stringWithoutWhitespaces = /.*\S.*/;
```

Comment:
this change seems to return a potential security issue in SonarCloud -> https://sonarcloud.io/project/security_hotspots?id=UK-Export-Finance_tfs-api&pullRequest=151&resolved=false&types=SECURITY_HOTSPOT

it may be worth investigating other options otherwise

Reply:
I wonder if a decorator would work better. something like this?
```
 @Transform(({ value }: TransformFnParams) => typeof value === "string" ? value.trim() : "" )
```
(where the Transformer is loaded from class-validator)

Reply:
There is actually a use case for searching with _leading_ spaces: the user might want to search for the string " abc" instead of just "abc". It's a narrow use case, but it doesn't require any extra work to support.

As for _trailing_ spaces, I tried it out and ACBS will error (with a 404) error if we try and send searchText with trailing spaces. Our code already removes trailing spaces anyway, but I've changed the validation to be "no spaces at the end of the string" so that our length validation still works correctly. Given that leading spaces are valid to search with, I think it would be confusing to the user if we silently remove trailing spaces so I prefer giving a bad request error instead.

Reply:
Oh, and the new regex doesn't have the potential security issue either!